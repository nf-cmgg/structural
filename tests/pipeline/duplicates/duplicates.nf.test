nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    tag "pipeline"
    tag "pipeline_duplicates"

    options "-stub"

    test("duplicate of 2 lines + 1 sample") {

        when {
            params {
                outdir = "${outputDir}"
                input = "${baseDir}/tests/pipeline/duplicates/samplesheet_duplicate2.csv"
                max_cpus   = 2
                max_memory = '6.GB'
                max_time   = '6.h'
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,
                    path("$outputDir").list().findAll{
                        !it.toString().endsWith("pipeline_info")
                    },
                    path("$outputDir").list().collect{
                        it.toString().split("/")[-1]
                    }
                ).match()}
            )
        }

    }

}


