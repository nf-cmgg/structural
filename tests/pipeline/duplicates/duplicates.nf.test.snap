{
    "duplicate of 2 lines + 1 sample": {
        "content": [
            [
                
            ],
            [
                [
                    "PosCon1.cnv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.cnv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.repeats.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.repeats.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.sv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.sv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.wisecondorx.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.wisecondorx_aberrations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.wisecondorx_bins.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.wisecondorx_chr_statistics.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.wisecondorx_segments.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr1.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr22.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "genome_wide.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                [
                    "PosCon3.cnv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.cnv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.repeats.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.repeats.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.sv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.sv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.wisecondorx.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.wisecondorx_aberrations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.wisecondorx_bins.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.wisecondorx_chr_statistics.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.wisecondorx_segments.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr1.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr22.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "genome_wide.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                "PosCon1",
                "PosCon3",
                "pipeline_info"
            ]
        ],
        "timestamp": "2023-11-30T13:19:08.640465122"
    }
}